src/EPR.SubsidiaryBulkUpload.Application/Services/BulkSubsidiaryProcessor.cs

```diff
@@ -1,24 +1,36 @@
 ﻿using EPR.SubsidiaryBulkUpload.Application.DTOs;
+using EPR.SubsidiaryBulkUpload.Application.Extensions;
 using EPR.SubsidiaryBulkUpload.Application.Models;
+using EPR.SubsidiaryBulkUpload.Application.Services.Interfaces;
 using Microsoft.Extensions.Logging;
 
 namespace EPR.SubsidiaryBulkUpload.Application.Services;
 
-public class BulkSubsidiaryProcessor(ISubsidiaryService organisationService, ICompaniesHouseDataProvider companiesHouseDataProvider, ILogger<BulkSubsidiaryProcessor> logger)
+public class BulkSubsidiaryProcessor(ISubsidiaryService organisationService, ICompaniesHouseDataProvider companiesHouseDataProvider, ILogger<BulkSubsidiaryProcessor> logger, INotificationService notificationService)
     : IBulkSubsidiaryProcessor
 {
+    private const string SubsidiaryBulkUploadMismatchedProgress = "Subsidiary bulk upload progress";
+    private const string SubsidiaryBulkUploadMismatchedErrors = "Subsidiary bulk upload errors. Mismatched subsidiaries.";
     private readonly ILogger<BulkSubsidiaryProcessor> _logger = logger;
     private readonly ISubsidiaryService organisationService = organisationService;
     private readonly ICompaniesHouseDataProvider companiesHouseDataProvider = companiesHouseDataProvider;
+    private readonly INotificationService _notificationService = notificationService;
 
-    public async Task Process(IEnumerable<CompaniesHouseCompany> subsidiaries, CompaniesHouseCompany parent, OrganisationResponseModel parentOrg, Guid userId)
+    public async Task Process(IEnumerable<CompaniesHouseCompany> subsidiaries, CompaniesHouseCompany parent, OrganisationResponseModel parentOrg, UserRequestModel userRequestModel)
     {
+        // these are the subsidiaries which exists in RPD
         var subsidiariesAndOrg = subsidiaries
             .ToAsyncEnumerable()
             .SelectAwait(async subsidiary => (Subsidiary: subsidiary, SubsidiaryOrg: await organisationService.GetCompanyByCompaniesHouseNumber(subsidiary.companies_house_number)));
 
-        // All subsidiaries with an org id, where no relationship already exists
-        var knownSubsidiariesToAdd = subsidiariesAndOrg.Where(co => co.SubsidiaryOrg != null)
+        // before going further. here we can check with RPD if the company name in the file is matching the RPD Org table company name
+        var subsidiariesAndOrgWithValidName = subsidiariesAndOrg
+            .Where(sub => sub.Subsidiary.companies_house_number == sub.SubsidiaryOrg.companiesHouseNumber && sub.Subsidiary.organisation_name == sub.SubsidiaryOrg.name);
+
+        // after checking the name only that collection will pass on for relationship check
```

Comment:
Comments need to be removed!

Reply:
sorry these comments are required for the logic understanding